reconcile/aus/base.py

```diff
@@ -1042,25 +1086,34 @@ def calculate_diff(
         list: upgrade policies to be applied
     """
 
-    def set_mutex(
-        locked: dict[str, str], cluster_id: str, mutexes: set[str] | None = None
+    locked: dict[str, str] = {}
+    sector_upgrades: dict[str, set[str]] = defaultdict(set)
+
+    def set_upgrading(
+        cluster_id: str, mutexes: set[str] | None, sector_name: str | None
     ) -> None:
         for mutex in mutexes or set():
             locked[mutex] = cluster_id
+        if sector_name:
+            sector_upgrades[sector_name].add(cluster_id)
 
     diffs: list[UpgradePolicyHandler] = []
 
     # all clusters IDs with a current upgradePolicy are considered locked
-    locked: dict[str, str] = {}
     for spec in desired_state.specs:
         if spec.cluster.id in [s.cluster.id for s in current_state]:
-            for mutex in spec.effective_mutexes:
-                locked[mutex] = spec.cluster.id
+            sector_name = spec.upgrade_policy.conditions.sector
+            set_upgrading(spec.cluster.id, spec.effective_mutexes, sector_name)
 
     addon_service = init_addon_service(desired_state.org.environment)
     now = datetime.utcnow()
     gates = get_version_gates(ocm_api)
     for spec in desired_state.specs:
+        sector_name = spec.upgrade_policy.conditions.sector
+        sector = None
+        if sector_name:
+            sector = desired_state.sectors[sector_name]
```

Comment:
this part is duplicated inside `verify_max_upgrades_should_skip`, can we do it once here and pass the result to verify?

```python
        sector_name = spec.upgrade_policy.conditions.sector
        sector = desired_state.sectors[sector_name] if sector_name else None # for typing right Sector | None
        ...
        if verify_max_upgrades_should_skip(spec, sector_upgrades, sector):
            continue
```

then for `verify_max_upgrades_should_skip`

```python
def verify_max_upgrades_should_skip(
    desired: ClusterUpgradeSpec,
    sector_upgrades: dict[str, set[str]],
    sector: Sector | None,
) -> bool:
    if sector is None:
        return False
    sector_name = sector.name
    ...
```